test/common/network/connection_impl_test.cc

```diff
@@ -1991,8 +1997,11 @@ class ReadBufferLimitTest : public ConnectionImplTest {
     listener_ = dispatcher_->createListener(socket_, listener_callbacks_, true);
 
     client_connection_ = dispatcher_->createClientConnection(
-        socket_->localAddress(), Network::Address::InstanceConstSharedPtr(),
-        Network::Test::createRawBufferSocket(), nullptr);
+        Utility::resolveUrl(fmt::format("tcp://{}:{}",
+                                        Network::Test::getLoopbackAddressUrlString(GetParam()),
+                                        socket_->localAddress()->ip()->port())),
```

Comment:
Does `socket_->localAddress()` on Win32 not return the listener address selected? Seems like we might depend on that elsewhere.

Reply:
For the "any" interface? Interesting question, I'll research.

Reply:
It appears this works (and if problems are encountered, localAddress() needs fixing.) Adopted the proposed change for this next commit.